pulp_rpm/tests/functional/api/test_sync.py

```diff
@@ -251,379 +150,103 @@ def test_file_decriptors(self):
+        keywords related to the reason of the failure. See :meth:`do_test`.
         """
-        # Step 1
-        repo = self.client.post(RPM_REPO_PATH, gen_repo())
-        self.addCleanup(self.client.delete, repo['pulp_href'])
-
-        # Step 2.
-
-        for body in [
-            gen_rpm_remote(),
-            gen_rpm_remote(url=RPM_SHA512_FIXTURE_URL),
-        ]:
-            remote = self.client.post(RPM_REMOTE_PATH, body)
-            self.addCleanup(self.client.delete, remote['pulp_href'])
-            # Sync the repository.
-            sync(self.cfg, remote, repo)
-
-        # Step 3
-        repo = self.client.get(repo['pulp_href'])
-        added_content = get_content(repo)[RPM_PACKAGE_CONTENT_NAME]
-        removed_content = get_removed_content(repo)[RPM_PACKAGE_CONTENT_NAME]
-
-        # In case of "duplicates" the most recent one is chosen, so the old
-        # package is removed from and the new one is added to a repo version.
-        self.assertEqual(len(added_content), RPM_PACKAGE_COUNT, added_content)
-        self.assertEqual(
-            len(removed_content), RPM_PACKAGE_COUNT, removed_content
-        )
-
-        # Verifying whether the packages with first checksum is removed and second
-        # is added.
-        self.assertEqual(added_content[0]['checksum_type'], 'sha512')
-        self.assertEqual(removed_content[0]['checksum_type'], 'sha256')
-
-
-class ChecksumConstraintTestCase(unittest.TestCase):
-    """Verify checksum constraint test case.
-
-    Do the following:
-
-    1. Create and sync a repo using the following
-       url=RPM_REFERENCES_UPDATEINFO_URL.
-    2. Create and sync a secondary repo using the following
-       url=RPM_REFERENCES_UPDATEINFO_URL.
-       Those urls have RPM packages with the same name.
-    3. Assert that the task succeed.
-
-    This test targets the following issue:
-
-    * `Pulp #4170 <https://pulp.plan.io/issues/4170>`_
-    * `Pulp #4255 <https://pulp.plan.io/issues/4255>`_
-    """
-
-    def test_sync(self):
-        """Test duplicate content can be synced."""
-        cfg = config.get_config()
-        client = api.Client(cfg)
-
-        for url in [RPM_REFERENCES_UPDATEINFO_URL, RPM_UNSIGNED_FIXTURE_URL]:
-            remote = client.post(RPM_REMOTE_PATH, gen_rpm_remote(url=url))
-            self.addCleanup(client.delete, remote['pulp_href'])
-
-            repo = client.post(RPM_REPO_PATH, gen_repo())
-            self.addCleanup(client.delete, repo['pulp_href'])
-
-            client.post(
-                urljoin(repo['pulp_href'], 'sync/'),
-                {'remote': remote['pulp_href']},
-            )
-            repo = client.get(repo['pulp_href'])
-
-            added_content_summary = get_added_content_summary(repo)
-            self.assertEqual(
-                added_content_summary,
-                RPM_FIXTURE_SUMMARY,
-                added_content_summary,
-            )
-
-
-class SyncModularContentTestCase(unittest.TestCase):
```

Comment:
we need to convert this test case also

Reply:
method 'test_sync_modular' in basic sync test class